design/reboot-interface.md

```diff
@@ -89,78 +89,58 @@ section of the BareMetalHost status. This records a time after which the Host
 was last booted using the current image. Processes running prior to this time
 may be assumed to have been stopped.
 
-To provide a declarative interface for performing a reboot, a new
-BareMetalHostRebootRequest CRD will be created. The spec of this custom
-resource will contain only a reference to a BareMetalHost. The status will
-contain timestamps for when the Host was powered off and on again. Once the
-power off time is present in the status, the host has been stopped. Each
-RebootRequest will be owned by the Host it refers to, so that if the Host is
-deleted all associated RebootRequest CRs will be cleaned up.
+A new date-time field, ``lastRebootRequest``, will be added to the ``provisioning``
+section of the BareMetalHost status. This records a time after which the Host
+was last requested to reboot (because we cannot trust any value from the user,
+who even if well intentioned, may have created the timestamp on a machine that 
+was not synchronised with the cluster or has a different timezone).
+
+Since the user interface requirements are still unclear, we will follow 
+standard practices of using an annotation ( ``host.metal3.io/reboot`` ) to 
+trigger reboots.  It shall contain the UID of the target Machine so that the
+controller can validate that the associated Machine has not changed since the
+annotation was created.
 
 The actual power management will be performed by the Host controller. This is
 necessary to avoid race conditions by ensuring that the ``Online`` flag and any
-reboot requests are managed in the same place. The Host controller
-will watch RebootRequest objects and trigger a reconciliation of the linked
-Host when one appears or changes.
-
-For simplicity, we will try to avoid creating a separate controller for the
-RebootRequest at all, but instead perform all operations on it in the course of
-reconciling the BareMetalHost object that it references. However, it may yet
-prove necessary to also create a trivial controller for the purposes of basic
-housekeeping (e.g. setting the owner of the resource).
-
-The creation timestamp of the request resource will be interpreted as the time
-of the reboot request. Requests that predate the last booted time will be
-immediately marked as complete without taking any further action. The Host
-controller will timestamp all pending reboot requests as having completed the
+reboot requests are managed in the same place. 
+
+If
+* the value in ``host.metal3.io/reboot`` matches the associated Machine, and
+* the value of ``lastRebootRequest`` is less than the ``lastPoweredOn``, and
```

Comment:
Shouldn't it be **greater** than?
if lastRebootRequest < lastPoweredOn than the host was already rebooted, isn't it?

Reply:
This section is saying that if `lastRebootRequest` < `lastPoweredOn` (meaning there has been no previous request since the host was booted/last rebooted), and the other conditions are met, then it will change `lastRebootRequest` to the current time meaning it will now be > `lastPoweredOn` and hence a reboot will commence.